drivers/hmis/app/graphql/mutations/add_household_members_to_enrollment.rb

```diff
@@ -0,0 +1,65 @@
+module Mutations
+  class AddHouseholdMembersToEnrollment < BaseMutation
+    argument :household_id, ID, required: true
+    date_string_argument :start_date, 'Start date with format yyyy-mm-dd', required: true
+    argument :household_members, [Types::HmisSchema::EnrollmentHouseholdMemberInput], required: true
+
+    field :enrollments, [Types::HmisSchema::Enrollment], null: true
+    field :errors, [Types::HmisSchema::ValidationError], null: false
+
+    def validate_input(household_id:, start_date:, household_members:)
+      errors = []
+      errors << InputValidationError.new('Entry date cannot be in the future', attribute: 'start_date') if Date.parse(start_date) > Date.today
+
+      has_hoh_enrollment = Hmis::Hud::Enrollment.viewable_by(current_user).exists?(household_id: household_id, relationship_to_ho_h: 1)
+      errors << InputValidationError.new("Cannot find Enrollment for household with id '#{household_id}'", attribute: 'household_id') unless has_hoh_enrollment
+      errors << InputValidationError.new('Enrollment already has a head of household designated', attribute: 'household_members') if has_hoh_enrollment && household_members.find { |hm| hm.relationship_to_ho_h == 1 }
+
+      errors
+    end
+
+    def resolve(household_id:, start_date:, household_members:)
+      user = current_user
+      errors = validate_input(household_id: household_id, start_date: start_date, household_members: household_members)
+
+      if errors.present?
+        return {
+          enrollments: [],
+          errors: errors,
+        }
+      end
+
+      hoh_enrollment = Hmis::Hud::Enrollment.viewable_by(user).find_by(household_id: household_id, relationship_to_ho_h: 1)
+      lookup = Hmis::Hud::Client.where(id: household_members.map(&:id)).map { |client| [client.id, client] }.to_h
+      project = hoh_enrollment.project
+
+      enrollments = household_members.map do |household_member|
+        enrollment = lookup[household_member.id.to_i].enrollments.viewable_by(user).find_by(household_id: household_id)
+
+        next enrollment if enrollment.present?
+
+        enrollment = Hmis::Hud::Enrollment.new(
+          data_source_id: hmis_user.data_source_id,
+          personal_id: lookup[household_member.id.to_i].personal_id,
+          relationship_to_ho_h: household_member.relationship_to_ho_h,
+          entry_date: Date.strptime(start_date, '%Y-%m-%d'),
+          project_id: project&.project_id,
+          household_id: household_id,
+          enrollment_id: SecureRandom.uuid.gsub(/-/, ''),
```

Comment:
`SecureRandom.uuid.gsub(/-/, '')` feels like something we should move up to the base class.  Something like:
```
def self.generate_id
  SecureRandom.uuid.gsub(/-/, '')
end
```
No need to fix now, but it's starting to feel like something we're going to see a bunch, and if we abstract it and change our minds on what to shove in there down the road, it'll be easy to update.

Reply:
@eanders That makes sense to me. I'll move it to methods on `Hmis::Hud::Enrollment`.

- [ ] Make this change

Reply:
I think one level up.  Every HUD table will need the same.

Reply:
@eanders So put it in `Hmis::Shared`?

Reply:
well, I was thinking `Hmis::Hud::Base`, but shared works too.